tests/vmi_configuration_test.go

```diff
@@ -287,6 +289,55 @@ var _ = Describe("[sig-compute]Configurations", decorators.SigCompute, func() {
 			),
 		)
 
+		It("with downwardMetrics virtio-serial", func() {
+			vmi := libvmi.NewCirros(libvmi.WithDownwardMetricsChannel())
+
+			vmi = tests.RunVMIAndExpectLaunch(vmi, 60)
+			libwait.WaitForSuccessfulVMIStart(vmi)
+
+			domXml, err := tests.GetRunningVirtualMachineInstanceDomainXML(virtClient, vmi)
+			Expect(err).ToNot(HaveOccurred())
+
+			var actualXML api.DomainSpec
+			err = xml.Unmarshal([]byte(domXml), &actualXML)
+			Expect(err).ToNot(HaveOccurred())
+
+			expectedXmlSpec := api.DomainSpec{
+				Devices: api.Devices{
+					Channels: []api.Channel{{
+						Type: "unix",
+						Source: &api.ChannelSource{
+							Mode: "bind",
+							Path: downwardmetrics.DownwardMetricsChannelSocket,
+						},
+						Target: &api.ChannelTarget{
+							Type: v1.VirtIO,
+							Name: "org.github.vhostmd.1",
+						},
+					}},
+				},
+			}
+
+			//Let's find the downwardMetrics channel index
+			i := 0
+			found := false
+			for j, channel := range actualXML.Devices.Channels {
+				if channel.Target.Name == expectedXmlSpec.Devices.Channels[0].Target.Name {
+					found = true
+					i = j
+					break
+				}
+			}
+			By("Check if the downwardMetric channel is present in the libvirt xml")
+			Expect(found).To(BeTrue())
+
+			Expect(expectedXmlSpec.Devices.Channels[0].Type).To(Equal(actualXML.Devices.Channels[i].Type))
+			Expect(expectedXmlSpec.Devices.Channels[0].Source.Mode).To(Equal(actualXML.Devices.Channels[i].Source.Mode))
```

Comment:
Would `Expect(expectedXmlSpec.Devices.Channels[0]).To(actualXML.Devices.Channels[i])` suffice?

Reply:
You are right. But we cannot be sure that `exectedXmlSpec.Devices.Channels[0].Target.Port` is going to be always `Port: 0`. This is something that libvirt decides. Therefore, IMHO, I think it's safer to leave it as it is.

Reply:
ACK, might as well just use constants for the asserted values then?

Reply:
Fixed